app/src/main/java/org/jellyfin/androidtv/browsing/StdGridFragment.java

```diff
@@ -131,15 +134,27 @@ public void onCreate(Bundle savedInstanceState) {
 
         mCardHeight = getCardHeight(mPosterSizeSetting);
 
-        setupUIElements();
+        if (mApplication.getUserPreferences().getGridDirection() == GridDirection.HORIZONTAL) {
+            HorizontalGridPresenter gridPresenter = new HorizontalGridPresenter();
+            gridPresenter.setNumberOfRows(getGridHeight() / getCardHeight());
+            setGridPresenter(gridPresenter);
+        }else {
+            VerticalGridPresenter gridPresenter = new VerticalGridPresenter();
+            // Why is this hardcoded you ask? Well did you ever look at getGridHeight()? Yup that one is hardcoded too
+            // This whole fragment is only optimized for 16:9 screens anyway
+            // is this bad? Yup it definitely is, we'll fix it when this screen is rewritten
+            // - Niels, 2020
+            gridPresenter.setNumberOfColumns(5);
```

Comment:
I think this does need to change based on the image size selected. For example small images look rather strange currently...

![small vertical grid](https://user-images.githubusercontent.com/3450688/82462085-229af880-9a89-11ea-8a11-e88c24e2d499.png)

Reply:
I _think_ I can add something for that but it will still be hardcoded so using a screen that's not 16:9 (or rather, 1920dp wide) or using images that are not poster-size (banners etc.) will make it glitch. This is unfortunately really hard to fix in this mess.